lib/reporter/stats-reporter.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+const colors = require('colors');
+const filesize = require('filesize');
+const Logger = require('../logger');
+const percent = require('calc-percent');
+const program = require('commander');
+const Table = require('cli-table');
+const uniq = require('lodash/uniq');
+
+const logger = new Logger();
+
+function statsReporter(statsObj) {
+	logger.updateLevel(program.logLevel);
+	const sizesPerType = getSizesPerType(statsObj.src, statsObj.dest);
+	const tableData = calculateSavingsPerType(sizesPerType);
+	const table = generateTable(tableData);
+
+	logger.log(table.toString());
+}
+
+function getSizesPerType(srcSizes, destSizes) {
+	const srcTypes = Object.keys(srcSizes);
+	const destTypes = Object.keys(destSizes);
+
+	return uniq(srcTypes.concat(destTypes)).reduce((sizes, type) => {
+		if (!srcSizes[type] || !destSizes[type]) return sizes;
+
+		sizes[type] = {
+			src: srcSizes[type],
+			dest: destSizes[type]
+		}
+
+		return sizes;
+	}, {});
+}
+
+function calculateSavingsPerType(sizes) {
+	return Object.keys(sizes).reduce((collection, type) => {
+		const savedBytes = sizes[type].src - sizes[type].dest;
+		collection[type] = Object.assign(sizes[type], { saving: savedBytes });
+
+		return collection;
+	}, {});
+}
+
+function generateTable(tableData) {
+	const table = new Table({
+		head: [
+			colors.blue('Filetype'),
+			colors.blue('Original size'),
+			colors.blue('Optimized size'),
+			colors.blue('Saving')
+		],
+		colAligns: ['', 'right', 'right', 'right'],
+		colWidths: [20, 20, 20, 20]
+	});
+
+	getTableRows(tableData)
+		.concat([getTotalRow(tableData)])
+		.map(row => table.push(row));
+
+	return table;
+}
+
+function getTableRows(tableData) {
+	return Object.keys(tableData).map(type => {
+		const sizes = tableData[type];
+		const row = [
+			type,
+			filesize(sizes.src),
+			filesize(sizes.dest),
+			filesize(sizes.saving) + ' (' + colors.green(percent(sizes.saving, [sizes.src], {suffix: '%'}))+ ')'
```

Comment:
if you change
```javascript
filesize(sizes.saving)
```
to
```javascript
const sizes = tableData[type];
const saving = calcSaving(sizes.src, sizes.dest)
```
you don't need the more complicated `calculateSavingsPerType()` anymore.